setup.py

```diff
@@ -30,44 +16,138 @@ def PyPiDescription():
 # $ pip install twine
 # $ pip install wheel
 # $ python setup.py bdist_wheel --universal
+
+# *** TEST YOUR PACKAGE WITH TEST PI ******
+# twine upload --repository-url https://test.pypi.org/legacy/ dist/*
+
+# If this is successful then push it to actual pypi
+
 # $ twine upload dist/*
 
-with open(os.path.join("mms", "version.py")) as f:
-    exec(f.read())
-
-requirements = ['Flask', 'Pillow', 'requests', 'flask-cors',
-                'psutil', 'jsonschema', 'onnx==1.1.1', 'boto3', 'importlib2',
-                'fasteners']
-# Enable Cu90 only when using linux with cuda enabled
-gpu_platform = False
-if platform.system().lower() == 'linux':
-    try:
-        # Check if CUDA is installed
-        cuda = ctypes.cdll.LoadLibrary('libcudart.so')
-        deviceCount = ctypes.c_int()
-        # get the number of supported GpUs
-        cuda.cudaGetDeviceCount(ctypes.byref(deviceCount))
-        if deviceCount.value > 0:
-            gpu_platform = True
-    except Exception as e:
-        gpu_platform = False
-if gpu_platform:
-    requirements = ['mxnet-cu90mkl>=1.2'] + requirements
-else:
-    requirements = ['mxnet-mkl>=1.2'] + requirements
-setup(
-    name='mxnet-model-server',
-    version=__version__.strip(),
-    description='Model Server for Apache MXNet is a tool for serving neural net models for inference',
-    long_description=PyPiDescription(),
-    url='https://github.com/awslabs/mxnet-model-server',
-    keywords='MXNet Model Server Serving Deep Learning Inference AI',
-    packages=pkgs,
-    install_requires=requirements,
-    entry_points={
-        'console_scripts': ['mxnet-model-server=mms.mxnet_model_server:start_serving',
-                            'mxnet-model-export=mms.export_model:export']
-    },
-    include_package_data=True,
-    license='Apache License Version 2.0'
-)
+
+import platform
+import os
+import ctypes
+import errno
+import time
+import sys
+from setuptools import setup, find_packages, Command
+import setuptools.command.build_py
+from shutil import copyfile
+import subprocess
+
+pkgs = find_packages(exclude=['*client_sdk_generator*', '*export_model*', '*swagger*', '*tests*'])
+source_server_file = os.path.abspath('frontend/server/build/libs/server-1.0.jar')
+dest_file_name = os.path.abspath('mms/frontend/model-server.jar')
+
+
+def pypi_description():
+    """Imports the long description for the project page"""
+    with open('PyPiDescription.rst') as df:
+        return df.read()
+
+
+def detect_model_server_version():
+    with open(os.path.abspath(os.path.join("mms", "version.py")), 'r') as vf:
+        exec(vf.read(), None, globals())
+    return __version__
+
+
+def select_mxnet(requirements=[]):
```

Comment:
This should be passed as parameter as well, should not depend on current host.

Also, "requirements" shadows variable in outer scope. You don't really need pass it in, you can append the return value at calling function.

Reply:
True. This can be removed.